packages/menu/src/SubMenu/SubMenu.spec.tsx

```diff
@@ -1,113 +1,288 @@
+        });
+        openMenu();
+
+        const subMenuA = getByTestId(subMenuTestId[0]);
+        userEvent.click(subMenuA);
+        const maybeSubMenuItem = queryByTestId(menuItemTestId[0]);
+        expect(maybeSubMenuItem).not.toBeInTheDocument();
+      });
+
+      test('with href does _not_ open submenu', () => {
+        const { getByTestId, queryByTestId, openMenu } = renderSubMenu();
+        openMenu();
 
-    // TODO: Fix redundant rendering in `Menu`. The submenu is closed on initial render, but opens on second render
-    // https://jira.mongodb.org/browse/LG-2904
-    waitFor(() => {
-      const subMenu = getByTestId(subMenu2Id);
-      expect(subMenu.tagName.toLowerCase()).toBe('a');
+        const subMenuB = getByTestId(subMenuTestId[1]);
+        userEvent.click(subMenuB);
+        const maybeSubMenuItem = queryByTestId(menuItemTestId[1]);
+        expect(maybeSubMenuItem).not.toBeInTheDocument();
+      });
+    });
+
+    describe('clicking the chevron button', () => {
+      test('opens the submenu', async () => {
+        const { getByTestId, openMenu } = renderSubMenu();
+        openMenu();
+        const subMenuB = getByTestId(subMenuTestId[1]);
+        const chevronB = globalGetByTestId(
+          subMenuB.parentElement!,
+          'lg-sub-menu-icon-button',
+        );
+
+        userEvent.click(chevronB);
+
+        const subMenuItemA = getByTestId(menuItemTestId[0]);
+        const subMenuItemB = getByTestId(menuItemTestId[1]);
+
+        await waitForElementToBeRemoved(subMenuItemA);
+        expect(subMenuItemB).toBeInTheDocument();
+      });
+
+      test('does _not_ fire a click handler', async () => {
+        const { getByTestId, openMenu } = renderSubMenu();
+        openMenu();
+        const subMenuA = getByTestId(subMenuTestId[0]);
+        const chevronA = globalGetByTestId(
+          subMenuA.parentElement!,
+          'lg-sub-menu-icon-button',
+        );
+
+        userEvent.click(chevronA);
+
+        expect(onClick).not.toHaveBeenCalled();
+      });
+
+      test('does _not_ navigate the window', async () => {
+        const { getByTestId, openMenu } = renderSubMenu();
+        openMenu();
+        const subMenuA = getByTestId(subMenuTestId[0]);
+        const chevronA = globalGetByTestId(
+          subMenuA.parentElement!,
+          'lg-sub-menu-icon-button',
+        );
+
+        userEvent.click(chevronA);
+
+        // Navigation is not implemented in jsdom
+        // So here we just check that the specific error was logged by jest
+        await waitFor(() => {
+          expect(errorSpy).not.toHaveBeenCalled();
+        });
+      });
+    });
+
+    test('Opening a Submenu closes the previous one', async () => {
+      const { getByTestId, queryByTestId, openMenu } = renderSubMenu();
+      openMenu();
+      const subMenuB = getByTestId(subMenuTestId[1]);
+      const chevronB = globalGetByTestId(
+        subMenuB.parentElement!,
+        'lg-sub-menu-icon-button',
```

Comment:
nit: this could also go in a variable

Reply:
I didn't put these in a variable because the value other ones don't really matter—they could be any string, as long as the string is consistent throughout the file. This id is defined in the component. I feel it's a bit confusing if changing some vars is safe, but changing others is not